client/src/pages/locations/Form.js

```diff
@@ -73,7 +83,6 @@ const LocationForm = ({
   const [attachmentList, setAttachmentList] = useState(
     initialValues?.attachments
   )
-  const [showSimilarLocations, setShowSimilarLocations] = useState(false)
```

Comment:
I would keep this line here, and add:
```javascript
  const [showSimilarLocationsMessage, setShowSimilarLocationsMessage] =
    useState(false)
  const [locationName, setLocationName] = useState(initialValues?.name)
```
immediately after it, so we have all state grouped together.
(The debounced callback declaration and the useEffect can stay where you added them.)